File: problems/字符串/14.最长公共前缀.py
#!/usr/bin/env python3
"""
14. 最长公共前缀
"""


def longest_common_prefix(strs: list[str]) -> str:
    if not strs:
        return ''
    for i in range(len(strs[0])):
        for j in range(1, len(strs)):
            if i >= len(strs[j]) or strs[j][i] != strs[0][i]:
                return strs[0][:i]
    return strs[0]


if __name__ == "__main__":
    # 测试用例
    print(longest_common_prefix(["flower", "flow", "flight"]))  # 预期: "fl"
    print(longest_common_prefix(["dog", "racecar", "car"]))      # 预期: ""
    print(longest_common_prefix(["a"]))                           # 预期: "a"

File: problems/字符串/14.最长公共前缀.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

string longestCommonPrefix(vector<string>& strs) {
    for (int i = 0; i < strs[0].size(); i++) {
        for (int j = 1; j < strs.size(); j++)
            if (i >= strs[j].size() || strs[j][i] != strs[0][i])
                return strs[0].substr(0, i);
    }
    return strs[0];
}

int main() {
    // 测试用例
    vector<string> t1 = {"flower", "flow", "flight"};
    cout << longestCommonPrefix(t1) << endl;  // 预期: fl
    vector<string> t2 = {"dog", "racecar", "car"};
    cout << longestCommonPrefix(t2) << endl;  // 预期: (空)
    vector<string> t3 = {"a"};
    cout << longestCommonPrefix(t3) << endl;  // 预期: a
    return 0;
}
